### PR TITLE
Fixed some foam dart bugs

### LINKED ISSUE #16084: Foam dart duplication bug?
HELP: Pecce/(Heimilich) (?) (PP) (VV) (SM) (FLW) (TP) (REJT): alright, so... http://puu.sh/nEOqA/2ac724732f.png on the right side of the cleanbot(?|F) was an assistant in a janicart, I shot at him with foam force crossbow and the darts seemed to duplicate. the darts you can see on the reinforced table are the result of this awesome bug, but I'm just letting you guys know about this in case you guys wanna do something about it.

### LINKED ISSUE #16975: Safety caps
1. Remove safety cap from foam force bullet
2. Load into empty gun
3. Fire gun
4. Safety cap is fired
5. No murder through toy accident :^)

### PR DESCRIPTION
**Pull request by Cruix**
Fixes #16975 
Fixes #16084

:cl:
bugfix: Fixed foam darts with the safety cap removed being reset when fired.
rscadd: Added the ability to remove pens from foam darts.
bugfix: Foam darts with pens inside them are no longer destroyed when fired.
bugfix: Foam darts no longer drop two darts when fired at a person on a janicart.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/reusable.dm [hunk 1]
```diff
@@ -38,6 +34,24 @@
 	icon_state = "foamdart"
 	ammo_type = /obj/item/ammo_casing/caseless/foam_dart
 	range = 10
+	var/obj/item/weapon/pen/pen = null
```
Comment (phil235) [inline code/modules/projectiles/projectile/reusable.dm]: the references must be deleted in /obj/item/projectile/bullet/reusable/foam_dart/Destroy()

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -10,6 +10,7 @@
 	pressure_resistance = INFINITY
 	var/def_zone = ""	//Aiming at
 	var/mob/firer = null//Who shot it
+	var/obj/item/ammo_casing/ammo_casing = null
```
Comment (phil235) [inline code/modules/projectiles/projectile.dm]: The reference must be deleted in obj/item/projectile/Destroy()

**Comment by Cruix** [comment]
Is it acceptable to qdel nulls, or should you generally check for null before qdeleting?

**Comment by phil235** [comment]
I think we usually check for null before, although the qdel() proc checks for it anyway.

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 3]
```diff
@@ -289,24 +289,41 @@
 	if (modified)
 		icon_state = "foamdart_empty"
 		desc = "Its nerf or nothing! ... Although, this one doesn't look too safe."
```
Comment (Iamgoofball) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: COPYRIGHT

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 4]
```diff
@@ -289,24 +289,41 @@
...
+		if(BB)
+			BB.icon_state = "foamdart_empty"
+	else
+		icon_state = "foamdart"
+		desc = "Its nerf or nothing! Ages 8 and up."
```
Comment (Iamgoofball) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: COPYRIGHT

Comment (coiax) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: I think you mean trademark. If you're going to kneejerk shut down ideas based on intellectual property law, then at least use the correct terminology.

**Pull request merged**